enterprise/admin/__init__.py

```diff
@@ -633,6 +634,59 @@ def get_form(self, request, obj=None, **kwargs):
         form.base_fields['content_filter'].initial = json.dumps(get_default_catalog_content_filter())
         return form
 
+    def get_actions(self, request):
+        """
+        Disallow the delete selected action as that does not send a DELETE request to enterprise-catalog
+        """
+        actions = super(EnterpriseCustomerCatalogAdmin, self).get_actions(request)
+        if 'delete_selected' in actions:
+            del actions['delete_selected']
+        return actions
+
+    def save_model(self, request, obj, form, change):
+        """
+        On save, creates or updates the corresponding catalog in the enterprise-catalog IDA.
+
+        `change` indicates whether or not the object is being updated (as opposed to created).
+        """
+        # Save the changes no matter what as we can copy them over later if we need to
+        super(EnterpriseCustomerCatalogAdmin, self).save_model(request, obj, form, change)
+
+        catalog_uuid = obj.uuid
+        catalog_client = EnterpriseCatalogApiClient(user=request.user)
+
+        if change:
+            update_fields = {
+                'enterprise_customer': str(obj.enterprise_customer.uuid),
+                'title': obj.title,
+                'content_filter': obj.content_filter,
+                'enabled_course_modes': obj.enabled_course_modes,
+                'publish_audit_enrollment_urls': obj.publish_audit_enrollment_urls,
+            }
+            catalog_client.update_enterprise_catalog(catalog_uuid, **update_fields)
+        else:
+            catalog_client.create_enterprise_catalog(
+                str(catalog_uuid),
+                str(obj.enterprise_customer.uuid),
+                obj.title,
+                obj.content_filter,
+                obj.enabled_course_modes,
+                obj.publish_audit_enrollment_urls,
+            )
+
+    def delete_model(self, request, obj):
+        """
+        Deletes the corresponding catalog in the enterprise-catalog IDA
+        """
+        catalog_uuid = obj.uuid
+        catalog_client = EnterpriseCatalogApiClient(user=request.user)
+        response = catalog_client.delete_enterprise_catalog(catalog_uuid)
+        # Early exit so we don't delete the catalog if the DELETE fails in enterprise-catalog
```

Comment:
A warning I just saw in the django docs says: 

> When overriding ModelAdmin.save_model() and ModelAdmin.delete_model(), your code must save/delete the object. They aren’t meant for veto purposes, rather they allow you to perform extra operations.

so it looks like we can't really guarantee the delete will be prevented. I don't know how often people actually delete enterprise catalogs to know if we're particularly concerned or if we're ok just checking the logs periodically and cleaning up any catalogs that don't get cleared out of the new service

Reply:
Based off Rob's answer on how often enterprise catalogs are deleted: 

> I know there have been times when they’ve been deleted, but as a team, we’ve made the determination that it is better not to delete catalogs

I feel comfortable allowing the deletion and then just handling any clean up afterwards since it should rarely (if ever) happen. @brittneyexline let me know if you disagree